crates/uv-python/src/discovery.rs

```diff
@@ -137,6 +137,7 @@ pub enum VersionRequest {
     MajorMinor(u8, u8),
     MajorMinorPatch(u8, u8, u8),
     MajorMinorPrerelease(u8, u8, Prerelease),
+    FreeThreaded(Box<VersionRequest>),
```

Comment:
Doing some internal debate about this right now — this is sort of the "easiest" way to add support but it might be better to added a `freethreaded: bool` to each relevant variant instead? Like `FreeThreaded(Any)` shouldn't be allowed nor `FreeThreaded(FreeThreaded(...))` right? Want to share some more about why you went with this pattern?

Reply:
It was at the time of implementing the easiest option to wrap every variant with a FreeThreaded version because it didn't felt right to clutter every variant with yet another flag. You are right that that `FreeThreaded(Any)` does not make sense but the `FreeThreaded(FreeThreaded(...))` should not be possible for the parser to generate, even though it would not be a very big problem because it would behave like a regular FreeThreaded(...) variant. But I understand the debate and would happily switch to a extra flag on every variant if you want me to.

Reply:
The parser can't generate it but this is a "public" enum so someone could construct that variant — which is a bit awkward. I think I have a preference for adding a boolean to the variants, it looks like it would make some of the downstream logic simpler. It is a bit sad that we'll need to add it everywhere we construct a `VersionRequest` manually but I'm not sure what else to do there. As a small note, it may need to be `Option<bool>` for cases in which the free-threaded build is fine, e.g., and can be used if it's first on the `PATH`.

Reply:
Fine by me. How would you parse a `3.13` request? Does the user allow a free-threaded build or not?

Reply:
I think generally no, unless it's the only one available — like we do for pre-releases #7300

We might want to be more strict than that but I'm not sure.

Reply:
I think you want to make sure this can change for a future Python version, but yes, I think currently you want free-threaded to be opt-in. In the future, if it works well, it might be opt-out instead, maybe as soon as 3.15 (which may be called 3.27).

Reply:
What if you have a `python` executable at the front of your path and it's free-threaded?

Reply:
If it's called `python` or `python3.13`, then yes, but usually it's only called `python3.13t`?

Reply:
Unfortunately people do weird things and we have to think about these edge-cases